pytorch_lightning/accelerators/gpu.py

```diff
@@ -53,6 +57,57 @@ def set_nvidia_flags(local_rank: int) -> None:
         devices = os.getenv("CUDA_VISIBLE_DEVICES", all_gpu_ids)
         _log.info(f"LOCAL_RANK: {local_rank} - CUDA_VISIBLE_DEVICES: [{devices}]")
 
+    def get_device_stats(self, device: Union[str, torch.device]) -> Dict[str, Any]:
+        """Gets stats for the given GPU device."""
+        if _TORCH_GREATER_EQUAL_1_8:
+            return torch.cuda.memory_stats(device)
+        else:
+            return self._get_gpu_stats(device)
+
+    def _get_gpu_stats(self, device: torch.device) -> Dict[str, float]:
+        nvidia_smi_path = shutil.which("nvidia-smi")
+        if nvidia_smi_path is None:
+            raise FileNotFoundError("nvidia-smi: command not found")
+
+        gpu_stat_keys = [
+            "utilization.gpu",
+            "memory.used",
+            "memory.free",
+            "utilization.memory",
+            "fan.speed",
+            "temperature.gpu",
+            "temperature.memoy",
+        ]
+        gpu_ids = self._get_gpu_id(device.index)
```

Comment:
my understanding from 

https://github.com/PyTorchLightning/pytorch-lightning/blob/f5608e90d6296fb448b55821b358b15e6cabaf4f/pytorch_lightning/core/mixins/device_dtype_mixin.py

was if device is GPU it will always be of type torch.device not str. But I am a bit unclear on this part - under what conditions would this be a str?